src/fast_radiology/models/unet.py

```diff
@@ -0,0 +1,111 @@
+""" Parts of the U-Net model """
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class DoubleConv(nn.Module):
+    """(convolution => [BN] => ReLU) * 2"""
+
+    def __init__(
+        self, in_channels, out_channels, mid_channels=None, with_dropout=False
+    ):
+        super().__init__()
+        if not mid_channels:
+            mid_channels = out_channels
+
+        conv1 = nn.Conv2d(in_channels, mid_channels, kernel_size=3, padding=1)
+        nn.init.kaiming_normal_(conv1.weight)
+        conv2 = nn.Conv2d(mid_channels, out_channels, kernel_size=3, padding=1)
+        nn.init.kaiming_normal_(conv2.weight)
+
+        seq = [conv1, nn.ReLU(inplace=True), conv2, nn.ReLU(inplace=True)]
+        if with_dropout:
+            seq.append(nn.Dropout(p=0.5))
+
+        self.double_conv = nn.Sequential(*seq)
+
+    def forward(self, x):
+        return self.double_conv(x)
+
+
+class Down(nn.Module):
+    """Downscaling with maxpool then double conv"""
+
+    def __init__(self, in_channels, out_channels, with_dropout=False):
+        super().__init__()
+        self.maxpool_conv = nn.Sequential(
+            nn.MaxPool2d(2),
+            DoubleConv(in_channels, out_channels, with_dropout=with_dropout),
+        )
+
+    def forward(self, x):
+        return self.maxpool_conv(x)
+
+
+class Up(nn.Module):
+    """Upscaling then double conv"""
+
+    def __init__(self, in_channels, out_channels):
+        super().__init__()
+        self.up = nn.ConvTranspose2d(
+            in_channels, in_channels // 2, kernel_size=2, stride=2
+        )
+        nn.init.kaiming_normal_(self.up.weight)
+        self.conv = DoubleConv(in_channels, out_channels)
+
+    def forward(self, x1, x2):
+        x1 = self.up(x1)
+        # input is CHW
+        diffY = x2.size()[2] - x1.size()[2]
+        diffX = x2.size()[3] - x1.size()[3]
+
+        x1 = F.pad(x1, [diffX // 2, diffX - diffX // 2, diffY // 2, diffY - diffY // 2])
```

Comment:
I was whether it is needed in our case or after we added padding, and not using `bilinear=True`, maybe we already have a proper size here..
But even if, that line would be obsolete instead of problematic.
I didn't reach that far in review in order to check size of tensor on that level